internal/app/machined/pkg/runtime/v1alpha1/board/rockpi4/rockpi4.go

```diff
@@ -0,0 +1,100 @@
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at http://mozilla.org/MPL/2.0/.
+
+package rock64
+
+import (
+	"fmt"
+	"io/ioutil"
+	"log"
+	"os"
+	"path/filepath"
+
+	"github.com/talos-systems/go-procfs/procfs"
+	"golang.org/x/sys/unix"
+
+	"github.com/talos-systems/talos/internal/app/machined/pkg/runtime"
+	"github.com/talos-systems/talos/pkg/copy"
+	"github.com/talos-systems/talos/pkg/machinery/constants"
+)
+
+var (
+	bin       = fmt.Sprintf("/usr/install/u-boot/%s/u-boot-rockchip.bin", constants.BoardRockpi4)
+	off int64 = 512 * 64
+	dtb       = "/dtb/rockchip/rk3399-rock-pi-4c.dtb"
+)
+
+// Rockpi4 represents the Radxa rock pi board.
+//
+// Reference: https://rockpi.org/
+type Rockpi4 struct{}
+
+// Name implements the runtime.Board.
+func (r *Rockpi4) Name() string {
+	return constants.BoardRockpi4
+}
+
+// Install implements the runtime.Board.
+func (r *Rockpi4) Install(disk string) (err error) {
+	var f *os.File
+
+	if f, err = os.OpenFile(disk, os.O_RDWR|unix.O_CLOEXEC, 0o666); err != nil {
+		return err
+	}
+	// nolint: errcheck
+	defer f.Close()
+
+	var uboot []byte
+
+	uboot, err = ioutil.ReadFile(bin)
```

Comment:
To avoid explicit declaration:
```suggestion
	uboot, err := ioutil.ReadFile(bin)
```

or

To check error on the same line:
```suggestion
	var uboot []byte

	if uboot, err = ioutil.ReadFile(bin); err != nil {
```

The same thing below